apps/ledger-live-desktop/src/renderer/components/SyncOnboarding/Manual/SyncOnboardingCompanion.tsx

```diff
@@ -351,14 +361,18 @@ const SyncOnboardingCompanion: React.FC<SyncOnboardingCompanionProps> = ({
     if (deviceOnboardingState?.isOnboarded && !seededDeviceHandled.current) {
       if (deviceOnboardingState?.currentOnboardingStep === DeviceOnboardingStep.Ready) {
         // device was just seeded
-        setStepKey(StepKey.Backup);
+        setStepKey(recoverUpsellRedirection?.enabled ? StepKey.Apps : StepKey.Backup);
```

Comment:
[SHOULD] Refacto to have one source of truth: 
`const firstStepKeyAfterSeeding = recoverUpsellRedirection?.enabled ? StepKey.Apps : StepKey.Backup` and use it here, and below, line 371.

Reply:
If redefining a variable, it's difficult to now what is this condition target to without seeing how the variable is defined.

Reply:
If your variable is named correctly, it should not be a problem

Reply:
The definition of correct is not the same for all persons, for example:
- `backupAfterSeed` should be negation when destructing;
- `recoverUpsellRedirectionEnabled`;
- `enableRecoverUpsellRedirection`;
- `firstStepKeyAfterSeeding` the one in the comment;
- more reasonable names.
So which one is the mostly correct ? Keep it original, test it and remove it later, better for all, I think.

Reply:
You know already using directly `recoverUpsellRedirection?.enabled` is not really respecting the separation of concerns. Why would this component have to worry about the upsell being enabled somewhere else?
If we wanted to do things really nicely, we would have a prop `isBackupStepEnabled: boolean` on this component, and the prop would be set to true/false by the parent, using `recoverUpsellRedirection?.enabled`.
The easier way to do that without touching the parents, is to define this boolean right at the top of the component and only rely on it for the internal logic of the component.
This component is already quite complicated and that's why I would avoid adding something very specific like `recoverUpsellRedirection` into the logic.

Reply:
I'm not sure what the separation of concerns you mentioned is;
This component does not worry about the FF activate/deactivate somewhere else, what I mean is, directly access to the FF can help us identify what the condition is and which FF can determine it;
Can not agree to "redefine a variable every time to store the property (`enabled`)" and I don't think it's a good idea because the name of this variable will be very different according to different developers, maybe this will cause confusion and the other developer must go back to the definition to know where (which FF) this boolean comes from;
Why this prop should be set in the parent component ? Introducing the FF where we need it is better;
About the 'easier way' sentence, I've explained above.
Yes, the component is complicated enough to add new logics, so using hook is a good solution, reading the FF { enabled } value to determine a component show or hide, simple and efficient and also good for further FF removal task.